model/project.go

```diff
@@ -26,8 +26,37 @@ const (
 	// DefaultCommandType is a system configuration option that is used to
 	// differentiate between setup related commands and actual testing commands.
 	DefaultCommandType = evergreen.CommandTypeTest
+
+	// Project permissions
+	PermissionProjectSettings  = "project_settings"
+	PermissionProjectVariables = "project_variables"
+	PermissionTasks            = "project_tasks"
+	PermissionPatches          = "project_patches"
+	PermissionLogs             = "project_logs"
 )
 
+var projectPermissions = []string{
+	PermissionProjectSettings,
+	PermissionProjectVariables,
+	PermissionTasks,
+	PermissionPatches,
+	PermissionLogs,
+}
+
+func init() {
+	env := evergreen.GetEnvironment()
+	if env == nil {
+		grip.Critical("environment not available")
+		return
+	}
+	rm := evergreen.GetEnvironment().RoleManager()
+	if rm == nil {
+		grip.Critical("role manager not available")
+		return
+	}
+	grip.Critical(rm.RegisterPermissions(projectPermissions))
```

Comment:
if we leave this all in the models package it should just panic on load. :)

conversely, we have a weird situation where this might end up panicing the agent?

Reply:
Yea I had to add a bunch of nil handling to avoid it :) I moved to the env now, and it's a lot cleaner